scripts/docker/dockerfile_gcc-12_cuda-12

```diff
@@ -41,22 +34,35 @@ USER ${USER}
 RUN git clone --recursive --branch $branch https://github.com/LLNL/axom.git axom_repo
 
 # Build/install TPLs via spack and then remove the temporary build directory on success
-RUN cd ${HOME}/axom_repo && python3 ./scripts/uberenv/uberenv.py --spack-env-file=./scripts/spack/configs/docker/ubuntu20_cuda/spack.yaml \
+RUN cd ${HOME}/axom_repo && python3 ./scripts/uberenv/uberenv.py --spack-env-file=./scripts/spack/configs/docker/ubuntu22_cuda/spack.yaml \
                                                                  --project-json=.uberenv_config.json \
-                                                                 --spec="%gcc@9.3.0+mfem+profiling+cuda cuda_arch=70" \
+                                                                 --spec="%gcc@12.3.0+mfem+profiling+cuda cuda_arch=70 ^raja+openmp+cuda ^umpire+openmp+cuda" \
                                                                  --prefix=${HOME}/axom_tpls -k \
                          && rm -rf ${HOME}/axom_tpls/build_stage ${HOME}/axom_tpls/spack
 
 # Make sure the new hostconfig works with a release build
 # Note: having high job slots causes build log to disappear and job to fail
 # Omit testing step, hangs at slam_lulesh unit test (same behavior for azure pipeline images, as well)
+# Disable link-time optimization from MPI wrapper causing linkage failures
 RUN cd ${HOME}/axom_repo && python3 config-build.py -hc *cuda.cmake \
                                                     -bp ${HOME}/axom_repo/build-release \
                                                     -ip ${HOME}/axom_repo/install-release \
+                                                    -DCMAKE_EXE_LINKER_FLAGS=-fno-lto  \
                                                     -bt Release \
                          && cd ${HOME}/axom_repo/build-release \
                          && make -j4 install
 
+# Install VisIt binary
+# Download tarball, installer, test script, perform installation and cleanup
+# Note - testing is done with command: ./visit/bin/visit -cli -nowin -s test_visit.py
+# (must be done manually; hangs when ran on Apple Silicon/ARM, x86_64/amd64 is okay)
+RUN wget https://github.com/visit-dav/visit/releases/download/v3.4.2/visit3_4_2.linux-x86_64-ubuntu22.tar.gz \
+    && wget https://github.com/visit-dav/visit/releases/download/v3.4.2/visit-install3_4_2 \
+    && wget https://raw.githubusercontent.com/visit-dav/visit/refs/heads/develop/scripts/docker/test_visit.py \
+    && chmod 777 visit-install3_4_2 \
+    && ./visit-install3_4_2 -c none 3.4.2 linux-x86_64-ubuntu22 ${HOME}/visit \
+    && rm -rf visit-install3_4_2 visit3_4_2.linux-x86_64-ubuntu22.tar.gz
```

Comment:
This change here installs the pre-built VisIt binary. I couldn't get the binary working on my Apple Silicon setup, but it did work on my personal Windows laptop: https://github.com/visit-dav/visit/issues/20532#issuecomment-3184429497